File: basic/cs205_c_cpp_2020s/lab03/rusrc/lab03_03.rs
mod candy_bar;

mod lab03_03 {
    use crate::candy_bar::candy_bar as CandyBar;

    pub(crate) fn main() {
        let cb_array = [
            CandyBar::CandyBar {
                name: String::from("Mona Coke"),
                weight: 3f64,
                calories: 31i64,
            },
            CandyBar::CandyBar {
                name: String::from("Mona Coke"),
                weight: 11.2f64,
                calories: 45i64,
            }, CandyBar::CandyBar {
                name: String::from("Mona Coke"),
                weight: 91.1f64,
                calories: 92i64,
            }
        ];
        for cb in cb_array.iter() {
            println!("{}", cb);
        }
    }
}

fn main() {
    lab03_03::main();
}

File: basic/cs205_c_cpp_2020s/lab03/exercise/lab03_03.cpp
/**
 * @Github: https://github.com/Certseeds
 * @Author: nanoseeds
 * @Date: 2020-02-28 09:09:46
 * @LastEditors: nanoseeds
 * @LastEditTime: 2021-02-09 08:50:49
 */
// SPDX-License-Identifier: AGPL-3.0-or-later
/*  CS205_C_CPP
    Copyright (C) 2020-2023 nanoseeds
*/
#include <iostream>
#include "CandyBar.h"

using std::cin;
using std::cout;
using std::endl;

int sub_main_03_03();

#ifndef UNIT_TESTING_LAB_03
#define UNIT_TESTING_LAB_03

int main() {
    const int32_t will_return = sub_main_03_03();
    cin.get();
    cin.get();
    return will_return;
}

#endif // !UNIT_TESTING_LAB_03

int sub_main_03_03() {
    const CandyBar cb_array[3]{
            {"Mona coke", 3,    31},
            {"Mona coke", 11.2, 45},
            {"Mona coke", 91.1, 92}
    };
    for (const auto &i: cb_array) {
        cout << "Brand: " << i.name << endl;
        cout << "Weight: " << i.weight << endl;
        cout << "Calorie: " << i.calories << endl;
    }
    return 0;
}
